tests/ci/worker/init_runner.sh

```diff
@@ -300,23 +300,46 @@ list_children () {
     echo "$children"
 }
 
+# https://github.com/actions/runner/issues/3266
+# We're unable to know if the runner is failed to start.
+# There's possibility that it fails because the runner's version is outdated,
+# so after the first failure we'll try to launch it with enabled autoupdate.
+#
+# We'll fail and terminate after 10 consequent failures.
+ATTEMPT=0
+
 while true; do
     runner_pid=$(pgrep Runner.Listener)
     echo "Got runner pid '$runner_pid'"
 
     if [ -z "$runner_pid" ]; then
+        echo Attempt $((++ATTEMPT)) to start the runner
         cd $RUNNER_HOME || terminate_and_exit
         detect_delayed_termination
         # If runner is not active, check that it needs to terminate itself
         echo "Checking if the instance suppose to terminate"
         no_terminating_metadata || terminate_on_event
         check_spot_instance_is_old && terminate_and_exit
         check_proceed_spot_termination force
+        # There were some failures to start the Job because of trash in _work
+        rm -rf _work
 
         echo "Going to configure runner"
-        sudo -u ubuntu ./config.sh --url $RUNNER_URL --token "$(get_runner_token)" \
-          --ephemeral --disableupdate --unattended \
-          --runnergroup Default --labels "$LABELS" --work _work --name "$INSTANCE_ID"
+        token_args=(--token "$(get_runner_token)")
+        config_args=(
+            "${token_args[@]}" --url "$RUNNER_URL"
+            --ephemeral --unattended --replace --runnergroup Default
+            --labels "$LABELS" --work _work --name "$INSTANCE_ID"
+        )
+        if (( ATTEMPT > 1 )); then
```

Comment:
I need to handle it correctly. Unfortunately, the `run.sh` does not provide information on whether the runner has finished successfully or failed. See https://github.com/actions/runner/issues/3266

So, there's the next plan:

- replace run.sh by `./bin/Runner.Listener` and write the last status into some temp file
- Read it and increase the ATTEMPT if last time it failed